### PR TITLE
SSI Templates for Header and Feedback form

### PR DESCRIPTION
**Pull request by deshmukhmayur**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

No common template/component for the header and feedback. Any SPA developer has to create their own. And very little to no consistency between the headers of different SPAs.

Issue Number: #20, #21 

## What is the new behavior?
A SPA developer can use the global components for header and feedback to use in their SPA.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Screenshots:
<details>
<summary>Show Screenshots:</summary>
<p>

- Header:
One Portal – Header

- Feedback Button:
One Portal – Feedback

- Feedback - Bug Report:
One Portal – Feedback - Bug Report

- Feedback - Share Feedback
One Portal – Feedback - Share Feedback

</p>
</details>

## Other information
- Edited the `.eslintrc.json` file to fix the tab indentations, and added a rule for enforcing semicolons.
- Added VS Code extensions and settings profile required for the syntax highlighting and formatting of the .ssi files and html string template literals.
- Added `docker-compose.yml` file with a service for SSI templates.

### CODE REVIEW

File: packages/ssi-service/src/feedback-panel/feedback.js [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+import { GraphQLClient } from 'graphql-request';
+import styles from './feedback.css';
+
+window.customElements.define('op-feedback', class extends HTMLElement {
+  constructor () {
+    super();
+
+    this.graphqlClient = new GraphQLClient('https://localhost:8080/graphql', {
+      headers: {
+        Authorization: 'Basic b25lX3BvcnRhbF9hcGk6PUw5dm5CK2RxNVoqcVo5cA=='
```
Comment (riginoommen) [inline packages/ssi-service/src/feedback-panel/feedback.js]: Credentials are Public. please use the environment variables also remove this from your commit history.

File: packages/ssi-service/src/feedback-panel/feedback.js [hunk 2]
```diff
@@ -0,0 +1,293 @@
+import html from 'html-template-tag';
+import { GraphQLClient } from 'graphql-request';
+import styles from './feedback.css';
+
+window.customElements.define('op-feedback', class extends HTMLElement {
+  constructor () {
+    super();
+
+    this.graphqlClient = new GraphQLClient('https://localhost:8080/graphql', {
```
Comment (riginoommen) [inline packages/ssi-service/src/feedback-panel/feedback.js]: If possible use the environment variables.

Comment (deshmukhmayur) [reply packages/ssi-service/src/feedback-panel/feedback.js]: Okay. Will probably add this in the build script.

File: packages/ssi-service/src/feedback-panel/feedback.js (hunk 1 above)
Comment (deshmukhmayur) [reply packages/ssi-service/src/feedback-panel/feedback.js]: @riginoommen This is just a Token. This shouldn't be much of a problem as long as the Authorisation backend has proper checks for it.
Also, this is not exactly a secret, as the client end js is a white box.

A solution to make this secure is to create a unique token for this purpose, and add limited access for it, to just submit a feedback/bug report.

Comment (sayak-sarkar) [inline packages/ssi-service/src/feedback-panel/feedback.js]: Why are the credentials publicly pushed here? Please remove this asap!

Comment (sayak-sarkar) [reply packages/ssi-service/src/feedback-panel/feedback.js]: Even if its just a token still it should NEVER be pushed to a public repo. Use ENV variables or any other appropriate approach. It's not just about security but also about practices.

Comment (deshmukhmayur) [reply packages/ssi-service/src/feedback-panel/feedback.js]: Removed it and added it in a `secrets.json` file

Comment (deshmukhmayur) [reply packages/ssi-service/src/feedback-panel/feedback.js]: Removed.

**Comment by deshmukhmayur** [comment]
Updated the PR with README and unit tests for the components

**Pull request merged**